src/TraceEvent/ETWReloggerTraceEventSource.cs

```diff
@@ -407,7 +407,7 @@ private unsafe void Initialize(TraceEventNativeMethods.EVENT_RECORD* rawData)
                         // first event (real time case).   This is really a problem, as we need that information, but we will go ahead and
                         // try to initialize as best we can. 
 
-                        m_source.pointerSize = sizeof(IntPtr);
+                        m_source.pointerSize = Environment.Is64BitOperatingSystem ? 8 : 4;
```

Comment:
I agree that pointerSize should be the underlying OS pointer size (we should never be using IntPtr (that is the size for the local process) to determine this value.    This is a good change.